internal/provider/resource_vault_cluster.go

```diff
@@ -0,0 +1,153 @@
+// before a cluster read operation should timeout.
+var defaultVaultClusterTimeout = time.Minute * 5
+
+// createUpdateTimeout is the amount of time that can elapse
+// before a cluster create or update operation should timeout.
+var createUpdateVaultClusterTimeout = time.Minute * 35
+
+// deleteTimeout is the amount of time that can elapse
+// before a cluster delete operation should timeout.
+var deleteVaultClusterTimeout = time.Minute * 25
+
+func resourceVaultCluster() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Vault cluster resource allows you to manage an HCP Vault cluster.",
+		CreateContext: resourceVaultClusterCreate,
+		ReadContext:   resourceVaultClusterRead,
+		UpdateContext: resourceVaultClusterUpdate,
+		DeleteContext: resourceVaultClusterDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &createUpdateVaultClusterTimeout,
+			Update:  &createUpdateVaultClusterTimeout,
+			Delete:  &deleteVaultClusterTimeout,
+			Default: &defaultVaultClusterTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"cluster_id": {
+				Description:      "The ID of the HCP Vault cluster.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_id": {
+				Description:      "The ID of the HVN this HCP Vault cluster is associated to.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// TODO: this is required for consul, but is this complete for vault yet? we can also mark computed output until it's ready.
+			"tier": {
+				Description: "The tier that the HCP Vault cluster will be provisioned as.  Only 'development' and 'standard' are available at this time.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				// TODO: generalize consul enum validator
+				// ValidateDiagFunc: validateStringInSlice(vaultClusterResourceTiers, true),
+				DiffSuppressFunc: func(_, old, new string, _ *schema.ResourceData) bool {
+					return strings.ToLower(old) == strings.ToLower(new)
+				},
+			},
+			// optional fields
+			"public_endpoint": {
+				Description: "Denotes that the cluster has a public endpoint for the Vault UI. Defaults to false.",
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"namespace": {
+				Description: "The name of the customer namespace.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				// TODO Confirm:
+				// ForceNew:    true,
+			},
+			"max_lease_ttl": {
+				Description: "The max lease time-to-live (TTL) for this Vault cluster.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				// TODO Confirm:
+				// ForceNew:    true,
+			},
+			// TODO: min_vault_version for consistency with Consul?
```

Comment:
`min_vault_version` or `initial_vault version`? or something else?

Reply:
I feel like `initial_vault_version` makes more sense semantically. But if it needs to change for consistency I'm sure we can do that.

Reply:
What does the Vault API call it today?

Reply:
It's called `initial_vault_version` in the Vault API. The question behind this question is, do we want HCP TF resources to be consistent with the service APIs, regardless of inconsistency across the services, or we do want to enforce some common language across the HCP provider as much as possible?

I might take a look at how the UI handles that problem.

Reply:
TODO: ticket to change `min_consul_version` on consul service to `initial_consul_version` for consistency.

Reply:
After some investigation on the consul side, we've decided to go with `min_vault_version` for consistency with the Consul clusters in the HCP and HCS provider. This way, the field can be used as be both the initial version of the cluster and to force version upgrades.